aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -654,6 +654,21 @@ TensorIterator TensorIterator::binary_op(Tensor& out, const Tensor& a,
   return iter;
 }
 
+TensorIterator TensorIterator::comparison_op(Tensor& out, const Tensor& a,
```

Comment:
Because this was originally in TH, there was no type promotion behavior. Since we've added some here, we should add a test for it.

Reply:
Added